src/main/java/net/consensys/orion/http/handler/receive/ReceiveHandler.java

```diff
@@ -79,12 +83,18 @@ public void handle(final RoutingContext routingContext) {
         return;
       }
 
-      final byte[] decryptedPayload;
-      try {
-        decryptedPayload = enclave.decrypt(encryptedPayload.get(), recipient);
-      } catch (final EnclaveException e) {
-        log.info("unable to decrypt payload with key {}", key);
-        routingContext.fail(404, new OrionException(OrionErrorCode.ENCLAVE_KEY_CANNOT_DECRYPT_PAYLOAD, e));
+      byte[] decryptedPayload = null;
+      EnclaveException exception = null;
+      for (final Box.PublicKey recipient : recipients) {
+        try {
+          decryptedPayload = enclave.decrypt(encryptedPayload.get(), recipient);
+        } catch (final EnclaveException e) {
+          exception = e;
```

Comment:
Anyway we can avoid assigning the exception to another variable like this?

Reply:
I think we can just remove it, as it does not really add value